### ISSUE TITLE
CPU crashing with 8 GB RAM and 2 vCPUs.

### ISSUE THREAD

**Issue by nikhilkawtakwar** [prisma/prisma#4892]
**Describe the bug**
we have a huge data set (thousands of records) and we do query a lot of data. After some nested queries got fired, we are getting 
```
prisma_1  | Uncaught error from thread [single-server-akka.actor.default-dispatcher-23]: GC overhead limit exceeded, shutting down JVM since 'akka.jvm-exit-on-fatal-error' is enabled for ActorSystem[single-server]
prisma_1  | java.lang.OutOfMemoryError: GC overhead limit exceeded
prisma_1  | 	at scala.collection.immutable.HashMap$HashMap1.get0(HashMap.scala:199)
prisma_1  | 	at scala.collection.immutable.HashMap$HashTrieMap.get0(HashMap.scala:326)
prisma_1  | 	at scala.collection.immutable.HashMap$HashTrieMap.get0(HashMap.scala:326)
prisma_1  | 	at scala.collection.immutable.HashMap.get(HashMap.scala:52)
prisma_1  | 	at scala.collection.MapLike.getOrElse(MapLike.scala:125)
prisma_1  | 	at scala.collection.MapLike.getOrElse$(MapLike.scala:125)
prisma_1  | 	at scala.collection.AbstractMap.getOrElse(Map.scala:59)
prisma_1  | 	at sangria.schema.ObjectLikeType.getField(Schema.scala:203)
prisma_1  | 	at sangria.schema.ObjectLikeType.getField$(Schema.scala:197)
prisma_1  | 	at sangria.schema.ObjectType.getField(Schema.scala:206)
prisma_1  | 	at sangria.execution.Resolver.isOptional(Resolver.scala:1178)
prisma_1  | 	at sangria.execution.Resolver.$anonfun$resolveActionsPar$29(Resolver.scala:730)
prisma_1  | 	at sangria.execution.Resolver$$Lambda$1822/52285303.apply(Unknown Source)
prisma_1  | 	at scala.collection.TraversableOnce.$anonfun$foldLeft$1(TraversableOnce.scala:156)
prisma_1  | 	at scala.collection.TraversableOnce.$anonfun$foldLeft$1$adapted(TraversableOnce.scala:156)
prisma_1  | 	at scala.collection.TraversableOnce$$Lambda$51/343856911.apply(Unknown Source)
prisma_1  | 	at scala.collection.Iterator.foreach(Iterator.scala:937)
prisma_1  | 	at scala.collection.Iterator.foreach$(Iterator.scala:937)
prisma_1  | 	at scala.collection.AbstractIterator.foreach(Iterator.scala:1425)
mysql_1   | 2019-09-23T07:16:30.012142Z 12 [Note] Aborted connection 12 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
prisma_1  | 	at scala.collection.IterableLike.foreach(IterableLike.scala:70)
prisma_1  | 	at scala.collection.IterableLike.foreach$(IterableLike.scala:69)
prisma_1  | 	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
prisma_1  | 	at scala.collection.TraversableOnce.foldLeft(TraversableOnce.scala:156)
prisma_1  | 	at scala.collection.TraversableOnce.foldLeft$(TraversableOnce.scala:154)
prisma_1  | 	at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
prisma_1  | 	at sangria.execution.Resolver.resolveActionsPar(Resolver.scala:729)
mysql_1   | 2019-09-23T07:16:30.012415Z 11 [Note] Aborted connection 11 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.012659Z 10 [Note] Aborted connection 10 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
prisma_1  | 	at sangria.execution.Resolver.resolveValue(Resolver.scala:901)
mysql_1   | 2019-09-23T07:16:30.012863Z 9 [Note] Aborted connection 9 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.013059Z 8 [Note] Aborted connection 8 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.013296Z 7 [Note] Aborted connection 7 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.013360Z 4 [Note] Aborted connection 4 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.013508Z 6 [Note] Aborted connection 6 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
mysql_1   | 2019-09-23T07:16:30.013558Z 2 [Note] Aborted connection 2 to db: 'prisma' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
prisma_1  | 	at sangria.execution.Resolver.resolveValue(Resolver.scala:807)
mysql_1   | 2019-09-23T07:16:30.013743Z 5 [Note] Aborted connection 5 to db: 'unconnected' user: 'root' host: '172.18.0.2' (Got an error reading communication packets)
```

which ends up in socket hang up error.
I saw similar kind of issue here in github however that was specific to mutation. We are getting this kind of error after querying for 3-4 level nested data. This might be duplicate but couldn't see any issue related to nested queries hence raising it.

**Screenshots**
Following screenshot show high usage when GC overhead limit exceeds
<img width="101" alt="Screenshot 2019-09-26 at 8 37 41 PM" src="https://user-images.githubusercontent.com/5787628/65700929-1b5c5000-e09e-11e9-987c-e2a47cb5f87f.png">

**Versions:**
 - Connector: `MySQL`
 - Prisma Server: `1.30.1`
 - `prisma` CLI: `prisma/1.30.1 (darwin-x64) node-v10.4.0`
 - OS: `Ubuntu 16.04`

**Comment by pantharshit00** [comment]
This GC bug was fixed with 1.34.7, please upgrade to that

**Comment by nikhilkawtakwar** [comment]
Ok Will do it. @pantharshit00 Thanks !
